api/src/main/java/jakarta/mail/internet/MimeMultipart.java

```diff
@@ -1002,7 +1003,13 @@ protected MimeBodyPart createMimeBodyPart(InternetHeaders headers,
      */
     protected MimeBodyPart createMimeBodyPart(InputStream is)
 				throws MessagingException {
-	return new MimeBodyPart(is);
+        if (ds != null && ds instanceof MimePartDataSource) {
```

Comment:
I think this should simply be:
`if (ds instanceof MessageAware)`

followed by a cast to MessageAware instead of MimePartDataSource